### ISSUE TITLE
optional `attention_mask` in `ORTModelForxxx`

### ISSUE THREAD

**Issue by IlyasMoutawwakil** [huggingface/optimum#1043]
### Feature request

The behavior of some `ORTModelForxxx` differs from that of `AutoModelForxxx`.
For example, `ORTModelForSequenceClassification` doesn't accept `None` as `attention_mask` even though it's an optional argument.
There's also some missing masks (`attention_mask`, `pixel_mask` or `head_mask` for example in audio and vision models) but I think those are intentionally omitted for now since they're not generated in exporters.

### Motivation

Drop in replacement when using `ORTModelForxxx` instead of `AutoModelForxxx`.

### Your contribution

I will submit a PR.
